ksqldb-parser/src/main/java/io/confluent/ksql/parser/properties/with/CreateSourceAsProperties.java

```diff
@@ -69,8 +69,12 @@ private CreateSourceAsProperties(final Map<String, Literal> originals) {
     props.validateDateTimeFormat(CommonCreateConfigs.TIMESTAMP_FORMAT_PROPERTY);
   }
 
+  public Optional<Format> getKeyFormat() {
+    return getKeyFormatInfo().map(FormatFactory::of);
+  }
```

Comment:
> I'm happy for you to leave this for another day. But if you have the time, rather than adding a new getKeyFormat method, maybe you could just remove the getValueFormat method and move the FormatFactory call inline. Then rename getValueFormatInfo to just getValueFormat.

Done.

> Also, getValueFormatProperties shouldn't be public. It's just API bloat. Better callers just call getValueFormat().getProperties().

`getValueFormat()` returns an Optional which may be empty, whereas there are uses of `getValueFormatProperties()` even if `getValueFormat()` is empty, for example: https://github.com/confluentinc/ksql/blob/0f04061f3b54d66e7beb820c846a7b950c43cd30/ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/Analyzer.java#L177-L179

Reply:
Agree that `getValueFormatProperties` should be public.  We'll have to ensure you can, for example, set the delimiter char even when the value/key format is not declared, but coming from system properties.  I wouldn't be surprised if there are already bugs in this area.